lstchain/reco/volume_reducer.py

```diff
@@ -0,0 +1,145 @@
+"""
+**TEMPORTAL** module with functions to perform a volume reduction to LST data:
+
+NOTE : The volume reduction SHOULD BE performed at the dl0 data levels.
+
+This is a temporal implementation of the 'tailcut and dilation' volume reduction method that computes
+the mask from the dl1 (integrated image) level, and applies it to the image (dl1) as well as to the waveform (dl0) and
+pulse_time arrays (dl1), filling the 'anti-mask' with zeros.
+
+This means that the 'first' dl0 computed data is OVERWRITTEN by the vol. reduced dl0 data.
+
+Usage:
+
+"import volume_reducer"
+
+In the configuration file:
+
+(...)
+
+"volume_reducer":{
+    "algorithm": "zero_suppression_tailcut_dilation",
+    "parameters": { # add here the desired tailcut parameters. See function's help for default parameters.
+    }
+ }
+
+(...)
+"""
+from ctapipe.image.cleaning import tailcuts_clean, dilate
+
+__all__ = ['check_volume_reduction_method',
+           'apply_volume_reduction',
+           'zero_suppression_tailcut_dilation'
+           ]
+
+
+def check_volume_reduction_method(config_file):
+    """
+    Checks in the configuration file if a volume reduction method has been set.
+
+    Parameters
+    ----------
+    config_file: dict
+        The configuration file used in the parent code.
+
+    Returns
+    -------
+    flag: bool
+        Flag indicating if the volume reducer method declared in the configuration file is(not) none
+    algorithm: str
+        Volume reduction algorithm name
+    """
+    if config_file['volume_reducer']['algorithm'] is None:
+        flag = False
+        algorithm = ''
+    else:
+        flag = True
+        algorithm = config_file['volume_reducer']['algorithm']
+        print(algorithm)
+
+    return flag, algorithm
+
+
+def apply_volume_reduction(event, tel_id, config):
+    """
+    Apply to an event the volume reduction method specified in the configuration file.
+
+    Parameters
+    ----------
+    event: 'ctapipe.io.containers.DataContainer'
+    tel_id: int
+        A telescope id wih data: an `event.r0.tels_with_data` object.
+    config: dict
+        Loaded configuration: configuration used to select the volume reducer method and its
+        corresponding parameters.
+
+    Returns
+    -------
+    none
+        Modifies the event container (at dl0) by applying the computed mask to the image, the waveform
+        and the pulse_time objects, as:
+            image[~mask] = 0, ...
+
+    """
+    camera = event.inst.subarray.tel[tel_id].camera
+
+    image = event.dl1.tel[tel_id].image  # Volume reduction mask computed, to date, at dl1 level !
+
+    pulse_time = event.dl1.tel[tel_id].pulse_time
+    waveform = event.dl0.tel[tel_id].waveform
+
+    algorithm = globals()[config['volume_reducer']['algorithm']]
```

Comment:
Here I think you should use the `get_volume_reduction_method` and if not `None` get the parameters and apply it.

Reply:
To get into this method, I have previously checked in the `dl0_to_dl1.py` script : 
`if volume_reduction_algorithm is not None:
                    apply_volume_reduction(...)`
So I will just pass the name of the algorithm (stored in the `volume_reduction_algorithm` variable rather than calling `get_volumen_reduction_method` that does the same).